models/Ion_auth_model.php

```diff
@@ -1141,12 +1153,16 @@ public function is_time_locked_out($identity) {
 	 * This function is no longer used.
 	 * It is kept for retrocompatibility purpose.
 	 *
-	 * @param	string $identity
-	 * @return	int
+	 * @param string $identity: user's identity
+	 * @param string $ip_address: IP address
+	 *                            Only used if track_login_ip_address set to TRUE.
+	 *                            If NULL (default value), current IP address is used.
+	 *                            Use get_last_attempt_ip($identity) to retrieve user's last IP
+	 * @return int
 	 */
-	public function get_last_attempt_time($identity) {
+	public function get_last_attempt_time($identity, $ip_address = NULL) {
 		if ($this->config->item('track_login_attempts', 'ion_auth')) {
-			$ip_address = $this->_prepare_ip($this->input->ip_address());
+			if (is_null($ip_address)) $ip_address = $this->_prepare_ip($this->input->ip_address());
```

Comment:
Can you re-format this to match the rest of the styling.  Also use isset instead of is_null in case the user sends a falsey value.  For formatting, either with curlys or as a ternary:

    if (!isset($ip_address)) 
    {
        $ip_address = $this->_prepare_ip($this->input->ip_address());
    }


or

    $ip_address =  isset($ip_address) ? $ip_address ? $this->_prepare_ip($this->input->ip_address());